app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -2,58 +2,60 @@
 
 import android.util.Log;
 
-import com.google.firebase.firestore.CollectionReference;
 import com.google.firebase.firestore.FirebaseFirestore;
 import com.google.firebase.firestore.FirebaseFirestoreSettings;
 
 import java.util.Map;
-import java.util.Random;
 
 public class DatabaseWrapper {
 
-    private static final String TAG = "DatabaseWrapper";
-    private static DatabaseWrapper INSTANCE = null;
-    private FirebaseFirestore firestore;
-
-    // final fields regarding ID generation
-    private static String ID_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
-    private static final int ID_LENGTH = 25;
-
-    private DatabaseWrapper() {
-        FirebaseFirestore.setLoggingEnabled(true);
-        FirebaseFirestoreSettings settings =
-                new FirebaseFirestoreSettings.Builder().setPersistenceEnabled(true).build();
-        try {
-            firestore = FirebaseFirestore.getInstance();
-            firestore.setFirestoreSettings(settings);
-        } catch (Exception e) {
-            Log.d(TAG, e.toString());
-        }
+  private static final String TAG = "DatabaseWrapper";
+  private static final int ID_LENGTH = 25;
+  private static DatabaseWrapper INSTANCE = null;
+  // final fields regarding ID generation
+  private static String ID_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  private FirebaseFirestore firestore;
+
+  private DatabaseWrapper() {
+    FirebaseFirestore.setLoggingEnabled(true);
+    FirebaseFirestoreSettings settings =
+        new FirebaseFirestoreSettings.Builder().setPersistenceEnabled(true).build();
+    try {
+      firestore = FirebaseFirestore.getInstance();
+      firestore.setFirestoreSettings(settings);
+    } catch (Exception e) {
+      Log.d(TAG, e.toString());
     }
+  }
 
-    public static DatabaseWrapper getInstance() {
-        if (INSTANCE == null) {
-            INSTANCE = new DatabaseWrapper();
-        }
-        return INSTANCE;
+  public static DatabaseWrapper getInstance() {
+    if (INSTANCE == null) {
+      INSTANCE = new DatabaseWrapper();
     }
+    return INSTANCE;
+  }
 
-    public static String generateRandomId() {
+  public static String generateRandomId() {
```

Comment:
Not relevant to this PR, but I'm assuming there was trouble with Java's Random package so we're doing the generation ourselves now right @muneebazam?

Reply:
yeah, the ID_CHARS is ugly but best we can do without using any external library